pomegranate/utils.pyx

```diff
@@ -118,8 +118,9 @@ cdef double gamma(double x) nogil:
 	cdef double den, num, result, z, y
 	cdef int i, n, arg_was_less_than_one
 
-	if x == 0.0:
-		return INF
+	with gil:
```

Comment:
That restriction seems fine. In general it is not okay to reacquire the GIL on low level functions like those in utils or the distributions because that will easily cascade into thousands of GIL acquiring/releasing which will mess with the multithreading. Doesn't the case below (x < 0.001) cover this case though?